src/build.d

```diff
@@ -1456,6 +1437,25 @@ class DependencyRef
     }
 }
 
+/**
+Takes a lambda and returns a memoized function to build a dependecy object.
+*/
+alias makeDep(alias Func) = memoize!(function() {
+    static struct DepBuilder
+    {
+        private DependencyRef dep;
+        auto ref opDispatch(string name, T)(T arg)
+        {
+            mixin("dep." ~ name ~ " = arg;");
```

Comment:
```suggestion
            mixin("dep.", name, " = arg;");
```

Reply:
oh I didn't know mixin supported that

Reply:
Hmmm, doesn't seem to work on the 2.087.0 compiler, and the autotester usese 2.079.0 so I don't think this will work for the time being.

Reply:
Looks like it's not supported till 2.083.1

Reply:
I forgot about that problem. You can mark as resolved.

Reply:
Thanks for letting me know about that feature.  I suggested it a few years back so good to see it's been added.

Reply:
It also works for `pragma(msg)` I believe

Reply:
Walter added it by refactoring `mixin(args)` to use the same handling as `pragma (msg, args)`, IIRC.